File: java/src/example/ExampleCustomize.java
package example;

import java.io.FileOutputStream;

import org.apache.poi.hssf.usermodel.HSSFWorkbook;
import org.apache.poi.xssf.usermodel.XSSFWorkbook;

import jp.co.systembase.core.DataTable;
import jp.co.systembase.report.Report;
import jp.co.systembase.report.ReportPages;
import jp.co.systembase.report.data.ReportDataSource;
import jp.co.systembase.report.renderer.pdf.PdfRenderer;
import jp.co.systembase.report.renderer.xls.XlsRenderer;
import jp.co.systembase.report.renderer.xlsx.XlsxRenderer;

// 機能サンプル 動的要素(デザイナのみ)
public class ExampleCustomize {

	public static void main(String[] args) throws Throwable {

		// カスタマイザを指定せずに、Reportオブジェクトを生成します
		Report report = new Report(ReadUtil.readJson("report/example_customize.rrpt"));
		report.fill(new ReportDataSource(getDataTable()));
		ReportPages pages = report.getPages();

		// PDF出力
		{
			FileOutputStream fos = new FileOutputStream("output/example_customize.pdf");
			try{
				pages.render(new PdfRenderer(fos));
			}finally{
				fos.close();
			}
		}

		// XLS出力
		{
			FileOutputStream fos = new FileOutputStream("output/example_customize.xls");
			try{
				HSSFWorkbook workBook = new HSSFWorkbook();
				XlsRenderer renderer = new XlsRenderer(workBook);
				renderer.newSheet("example_render");
				pages.render(renderer);
				workBook.write(fos);
			}finally{
				fos.close();
			}
		}

		// XLSX出力
		{
			FileOutputStream fos = new FileOutputStream("output/example_customize.xlsx");
			try{
				XSSFWorkbook workBook = new XSSFWorkbook();
				XlsxRenderer renderer = new XlsxRenderer(workBook);
				renderer.newSheet("example_render");
				pages.render(renderer);
				workBook.write(fos);
			}finally{
				fos.close();
			}
		}
	}

	private static DataTable getDataTable(){
		DataTable ret = new DataTable();
		ret.setFieldNames("NUM");
		ret.addRecord().puts(50);
		ret.addRecord().puts(40);
		ret.addRecord().puts(30);
		ret.addRecord().puts(20);
		ret.addRecord().puts(10);
		ret.addRecord().puts(0);
		ret.addRecord().puts(-10);
		ret.addRecord().puts(-20);
		ret.addRecord().puts(-30);
		ret.addRecord().puts(-40);
		ret.addRecord().puts(-50);
		return ret;
	}

}

File: cs_core/example/ExampleCustomize.cs
using System;
using System.IO;
using System.Data;

using jp.co.systembase.NPOI.XSSF.UserModel;

using jp.co.systembase.json;
using jp.co.systembase.report;
using jp.co.systembase.report.data;
using jp.co.systembase.report.renderer.pdf;
using jp.co.systembase.report.renderer.xlsx;

// 機能サンプル 動的要素(デザイナのみ)
namespace example
{
    class ExampleCustomize
    {

        public static void Run()
        {
            // カスタマイザを指定せずに、Reportオブジェクトを生成します
            Report report = new Report(Json.Read("report/example_customize.rrpt"));
            report.Fill(new ReportDataSource(getDataTable()));
            ReportPages pages = report.GetPages();

            // PDF出力
            using (FileStream fs = new FileStream("output/example_customize.pdf", FileMode.Create))
            {
                PdfRenderer renderer = new PdfRenderer(fs);
                renderer.Setting.ReplaceBackslashToYen = true;
                pages.Render(renderer);
            }

            // XLSX出力
            using (FileStream fs = new FileStream("output/example_customize.xlsx", FileMode.Create))
            {
                XSSFWorkbook workbook = new XSSFWorkbook();
                XlsxRenderer renderer = new XlsxRenderer(workbook);
                renderer.NewSheet("example_customize");
                pages.Render(renderer);
                workbook.Write(fs);
            }
        }

        private static DataTable getDataTable()
        {
            DataTable ret = new DataTable();
            ret.Columns.Add("NUM", typeof(Decimal));
            ret.Rows.Add(50);
            ret.Rows.Add(40);
            ret.Rows.Add(30);
            ret.Rows.Add(20);
            ret.Rows.Add(10);
            ret.Rows.Add(0);
            ret.Rows.Add(-10);
            ret.Rows.Add(-20);
            ret.Rows.Add(-30);
            ret.Rows.Add(-40);
            ret.Rows.Add(-50);
            return ret;
        }

    }

}